crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1238 @@
+        result_list.append(self.acquire_alignment)?;
+        result_list.append(self.qubit_properties.clone())?;
+        result_list.append(self.concurrent_measurements.clone())?;
+        result_list.append(
+            self.gate_map
+                .clone()
+                .into_iter()
+                .map(|(key, value)| {
+                    (
+                        key,
+                        value
+                            .into_iter()
+                            .collect::<Vec<(Option<Qargs>, Option<BaseInstructionProperties>)>>(),
+                    )
+                })
+                .collect::<GateMapState>()
+                .into_py(py),
+        )?;
+        result_list.append(self._gate_name_map.clone().into_py(py))?;
+        result_list.append(self.global_operations.clone())?;
+        result_list.append(self.qarg_gate_map.clone().into_iter().collect_vec())?;
+        result_list.append(self.non_global_basis.clone())?;
+        result_list.append(self.non_global_strict_basis.clone())?;
+        Ok(result_list.to_owned().unbind())
+    }
+
+    fn __setstate__(&mut self, state: Bound<PyList>) -> PyResult<()> {
+        self.description = state.get_item(0)?.extract::<Option<String>>()?;
+        self.num_qubits = state.get_item(1)?.extract::<Option<usize>>()?;
+        self.dt = state.get_item(2)?.extract::<Option<f64>>()?;
+        self.granularity = state.get_item(3)?.extract::<i32>()?;
+        self.min_length = state.get_item(4)?.extract::<usize>()?;
+        self.pulse_alignment = state.get_item(5)?.extract::<i32>()?;
+        self.acquire_alignment = state.get_item(6)?.extract::<i32>()?;
+        self.qubit_properties = state.get_item(7)?.extract::<Option<Vec<PyObject>>>()?;
+        self.concurrent_measurements = state.get_item(8)?.extract::<Vec<Vec<usize>>>()?;
+        self.gate_map = IndexMap::from_iter(
+            state
+                .get_item(9)?
+                .extract::<GateMapState>()?
+                .into_iter()
+                .map(|(name, prop_map)| (name, IndexMap::from_iter(prop_map.into_iter()))),
+        );
+        self._gate_name_map = state.get_item(10)?.extract::<IndexMap<String, GateRep>>()?;
+        self.global_operations = state
+            .get_item(11)?
+            .extract::<IndexMap<usize, HashSet<String>>>()?;
+        self.qarg_gate_map = IndexMap::from_iter(
+            state
+                .get_item(12)?
+                .extract::<Vec<(Option<Qargs>, Option<HashSet<String>>)>>()?,
+        );
+        self.non_global_basis = state.get_item(13)?.extract::<Option<Vec<String>>>()?;
+        self.non_global_strict_basis = state.get_item(14)?.extract::<Option<Vec<String>>>()?;
+        Ok(())
+    }
+}
+
+// Rust native methods
+impl BaseTarget {
+    /// Generate non global operations if missing
+    fn generate_non_global_op_names(&mut self, strict_direction: bool) -> &Vec<String> {
+        let mut search_set: HashSet<Option<Qargs>> = HashSet::new();
```

Comment:
With `None` there can't be any other constraints on it because that means it's globally defined. This probably can be simplified if you changed search set to just be HashSet<Qargs> and you detect and  `None` and skip it.